### PR TITLE
Implement selenium test for covering LS server for TypeScript language

### PR DESCRIPTION
**Pull request by musienko-maxim**
### What does this PR do?
This PR implements simple scenario: create a workspace with enabled TypeScript LS installer, open simple TypeScript project, check initialization TypeScript LS by UI, checks code validation, autocompletion, go to definition features.

### What issues does this PR fix or reference?
https://github.com/eclipse/che/issues/9918

### CODE REVIEW

**Review by artaleks9 — APPROVED** [review]

File: selenium/che-selenium-test/src/main/java/org/eclipse/che/selenium/pageobject/CodenvyEditor.java [hunk 1]
```diff
@@ -794,6 +794,19 @@ public void waitMarkerInPositionAndClick(MarkerLocator markerLocator, int positi
     seleniumWebDriverHelper.waitAndClick(By.xpath(format(markerLocator.get(), position)));
   }
 
+  /**
+   * Waits marker with specified {@code markerLocator} on the defined {@code position} and move
```
Comment (dmytro-ndp) [inline selenium/che-selenium-test/src/main/java/org/eclipse/che/selenium/pageobject/CodenvyEditor.java]: `position` could confuse a little so as it's about the `line` number, not just about specific position in line.

File: selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/languageserver/TypeScriptEditingTest.java [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+            "Finished language servers initialization, file path '/%s'", PATH_TO_GREETER_FILE);
+
+    consoles.waitExpectedTextIntoConsole(intitTypeScriptLanguageServerMessage);
+    checkCodeValidation();
+    checkCodeAssistant();
+    checkGoToDefinition();
+  }
+
+  private void checkCodeValidation() {
+    final int expectedValueOfErrorMarkers = 9;
```
Comment (dmytro-ndp) [inline selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/languageserver/TypeScriptEditingTest.java]: from the readability point it's better to divide verification by given/when/then sections.

File: selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/languageserver/TypeScriptEditingTest.java [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+            expectedValueOfErrorMarkers, actualValueErrorMarkers));
+    editor.waitMarkerInPositionAndMoveCursor(ERROR_OVERVIEW, 13);
+    editor.waitTextInToolTipPopup("Cannot find name 'c'");
+    editor.goToPosition(13, 2);
+    editor.typeTextIntoEditor(DELETE.toString());
+    editor.waitAllMarkersInvisibility(ERROR);
+  }
+
+  private void checkCodeAssistant() {
+    String textFromWholeCodeAssistantScope =
```
Comment (dmytro-ndp) [inline selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/languageserver/TypeScriptEditingTest.java:113]: from the readability point it's better to divide verification by given/when/then sections

File: selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/languageserver/TypeScriptEditingTest.java [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+        "AbortController\nAbortSignal\nabstract\nActiveXObject\naddEventListener\nalert";
+    String textFromGreeterObject = "greet\ngreeting\ntestPrint";
+
+    editor.goToPosition(28, 36);
+    editor.launchAutocomplete();
+    editor.waitTextIntoAutocompleteContainer(textFromWholeCodeAssistantScope);
+    editor.closeAutocomplete();
+    editor.typeTextIntoEditor(ENTER.toString());
+    editor.typeTextIntoEditor("greeter.");
```
Comment (dmytro-ndp) [inline selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/languageserver/TypeScriptEditingTest.java]: it's hard to understand test case here
Could you, please, name string "greeter." somehow to make its propose an obvious?

File: selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/languageserver/TypeScriptEditingTest.java [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+
+package org.eclipse.che.selenium.languageserver;
+
+import static org.eclipse.che.selenium.core.constant.TestMenuCommandsConstants.Assistant.ASSISTANT;
+import static org.eclipse.che.selenium.core.constant.TestMenuCommandsConstants.Assistant.FIND_DEFINITION;
+import static org.eclipse.che.selenium.core.project.ProjectTemplates.NODE_JS;
+import static org.eclipse.che.selenium.core.workspace.WorkspaceTemplate.ECLIPSE_NODEJS;
+import static org.eclipse.che.selenium.pageobject.CodenvyEditor.MarkerLocator.ERROR;
+import static org.eclipse.che.selenium.pageobject.CodenvyEditor.MarkerLocator.ERROR_OVERVIEW;
+import static org.openqa.selenium.Keys.*;
```
Comment (Ohrimenko1988) [inline selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/languageserver/TypeScriptEditingTest.java]: This is a kind of anti-pattern, please use direct static import

File: selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/languageserver/TypeScriptEditingTest.java [hunk 6]
```diff
@@ -0,0 +1,128 @@
...
+    editor.typeTextIntoEditor(SPACE.toString());
+    final int actualValueErrorMarkers = editor.getMarkersQuantity(ERROR);
+    assertEquals(
+        actualValueErrorMarkers,
+        expectedValueOfErrorMarkers,
+        String.format(
+            "The expected value of errors marker should be %d but actual %d",
+            expectedValueOfErrorMarkers, actualValueErrorMarkers));
+    editor.waitMarkerInPositionAndMoveCursor(ERROR_OVERVIEW, 13);
+    editor.waitTextInToolTipPopup("Cannot find name 'c'");
```
Comment (dmytro-ndp) [inline selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/languageserver/TypeScriptEditingTest.java]: it's hard to understand test case here
Could you, please, name string "Cannot find name 'c'" somehow to make its propose an obvious?

File: selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/languageserver/TypeScriptEditingTest.java [hunk 7]
```diff
@@ -0,0 +1,128 @@
...
+    editor.typeTextIntoEditor(ENTER.toString());
+    editor.typeTextIntoEditor("greeter.");
+    editor.launchAutocomplete();
+    editor.waitTextIntoAutocompleteContainer(textFromGreeterObject);
+    editor.selectItemIntoAutocompleteAndPerformDoubleClick("greet");
+    editor.waitAllMarkersInvisibility(ERROR);
+  }
+
+  private void checkGoToDefinition() {
+    editor.goToPosition(24, 20);
```
Comment (dmytro-ndp) [inline selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/languageserver/TypeScriptEditingTest.java]: Could you, please, comment in the code which definition we a going to here?

File: selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/languageserver/TypeScriptEditingTest.java (hunk 6 above)
Comment (musienko-maxim) [reply selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/languageserver/TypeScriptEditingTest.java]: This message is returned by Language Server and we can not control this one in this case

File: selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/languageserver/TypeScriptEditingTest.java (hunk 2 above)
Comment (Ohrimenko1988) [reply selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/languageserver/TypeScriptEditingTest.java]: Maybe "expectedCountOfErrorMarkers" better?

**Review by Ohrimenko1988 — APPROVED** [review]

File: selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/languageserver/TypeScriptEditingTest.java (hunk 4 above)
Comment (musienko-maxim) [reply selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/languageserver/TypeScriptEditingTest.java]: In the test based on simple Greeter example like:
[Greeter example](url)
```typescript
class Greeter {
    greeting: string;
    constructor(message: string) {
        this.greeting = message;
    }
    greet() {
        return "Hello, " + this.greeting;
    }
    
     testPrint(): void {
        const printVar = new Print();
        printVar.print("test print");
    }
}

let greeter = new Greeter("world");
```
and i cannot see difficult parts for understanding

File: selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/languageserver/TypeScriptEditingTest.java (hunk 6 above)
Comment (dmytro-ndp) [reply selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/languageserver/TypeScriptEditingTest.java]: I didn't mean control - I meant name this string somehow to make its propose clear, and we can do it in our test.

File: selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/languageserver/TypeScriptEditingTest.java (hunk 4 above)
Comment (musienko-maxim) [reply selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/languageserver/TypeScriptEditingTest.java]: Ok, your proposal?

File: selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/languageserver/TypeScriptEditingTest.java (hunk 6 above)
Comment (musienko-maxim) [reply selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/languageserver/TypeScriptEditingTest.java]: OK, agree

File: selenium/che-selenium-test/src/main/java/org/eclipse/che/selenium/pageobject/CodenvyEditor.java [hunk 8]
```diff
@@ -794,6 +794,19 @@ public void waitMarkerInPositionAndClick(MarkerLocator markerLocator, int positi
...
   }
 
+  /**
+   * Waits marker with specified {@code markerLocator} on the defined {@code line number} and move
+   * cursor to it
+   *
+   * @param markerLocator marker's type, defined in {@link MarkerLocator}
+   * @param position line's number, where marker is expected
+   */
+  public void waitMarkerInPositionAndMoveCursor(MarkerLocator markerLocator, int position) {
```
Comment (dmytro-ndp) [inline selenium/che-selenium-test/src/main/java/org/eclipse/che/selenium/pageobject/CodenvyEditor.java]: `moveCursorToLineWithMarker(MarkerLocator markerLocator, int line)` would explain the propose of method better, IMHO

File: selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/languageserver/TypeScriptEditingTest.java [hunk 9]
```diff
@@ -0,0 +1,140 @@
...
+            expectedAmountOfErrorMarkers, actualValueErrorMarkers));
+    editor.waitMarkerInPositionAndMoveCursor(ERROR_OVERVIEW, 13);
+    editor.waitTextInToolTipPopup(tooltipWithErrorMessage);
+    editor.goToPosition(13, 2);
+    editor.typeTextIntoEditor(DELETE.toString());
+    editor.waitAllMarkersInvisibility(ERROR);
+  }
+
+  private void checkCodeAssistant() {
+
```
Comment (dmytro-ndp) [inline selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/languageserver/TypeScriptEditingTest.java]: redundant empty line

File: selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/languageserver/TypeScriptEditingTest.java [hunk 10]
```diff
@@ -0,0 +1,140 @@
...
+    String textFromWholeCodeAssistantScope =
+        "AbortController\nAbortSignal\nabstract\nActiveXObject\naddEventListener\nalert";
+
+    String textFromGreeterObject = "greet\ngreeting\ntestPrint";
+
+    String nameOfGreeterClassRef = "greeter.";
+
+    editor.goToPosition(28, 36);
```
Comment (dmytro-ndp) [inline selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/languageserver/TypeScriptEditingTest.java]: could you, please divide separate steps by empty lines to improve readability?

File: selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/languageserver/TypeScriptEditingTest.java (hunk 3 above)
Comment (musienko-maxim) [reply selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/languageserver/TypeScriptEditingTest.java:113]: I have not quite understood what do you mean?
use something like that:
```java
 // when
    projectExplorer.openItemByPath(PATH_TO_PROGRAM);
    editor.setInactiveBreakpoint(22);
    editor.closeAllTabs();
    commandsPalette.openCommandPalette();
    commandsPalette.startCommandByDoubleClick(MAKE_AND_DEBUG_COMMAND_NAME);
    consoles.waitExpectedTextIntoConsole("Listening on port 8001");

    menu.runCommandByXpath(
        TestMenuCommandsConstants.Run.RUN_MENU,
        TestMenuCommandsConstants.Run.DEBUG,
        getXpathForDebugConfigurationMenuItem());

    notifications.waitExpectedMessageOnProgressPanelAndClosed(
        String.format("Remote debugger connected\nConnected to: localhost:%s.", DEBUG_PORT));

    // then
    editor.waitTabFileWithSavedStatus("hello.cc");
    debugPanel.waitDebugHighlightedText("  return \"Hello World, \" + name + \"!\";");
    debugPanel.waitTextInVariablesPanel("name =");
    debugPanel.waitTextInVariablesPanel("\"man\"");
```

Comment (dmytro-ndp) [reply selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/languageserver/TypeScriptEditingTest.java:113]: yes

**Review by dmytro-ndp — APPROVED** [review]

**Comment by musienko-maxim** [comment]
ci-build

**Comment by musienko-maxim** [comment]
ci-test

**Pull request merged**